src/hubAuth.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as path from 'path';
+import * as os from 'os';
+import { fs } from '@salesforce/core';
+import * as shell from 'shelljs';
+import { debug } from 'debug';
+
+// this seems to be a known eslint error for enums
+// eslint-disable-next-line no-shadow
+enum AuthStrategy {
+  JWT = 'JWT',
+  AUTH_URL = 'AUTH_URL',
+  REUSE = 'REUSE',
+  NONE = 'NONE',
+}
+
+export const authFromStubbedHome = (homeDir: string): void => {
+  const logger = debug('testkit:authFromStubbedHome');
+  if (getAuthStrategy() === AuthStrategy.JWT && process.env.TESTKIT_JWT_KEY) {
+    logger('trying jwt auth');
+    const jwtKey = path.join(homeDir, 'jwtKey');
+    fs.writeFileSync(jwtKey, process.env.TESTKIT_JWT_KEY);
```

Comment:
Why does the key file need to be copied?

Reply:
it's probably read from an env in CI?

Reply:
In Jenkins, one can store a "secret file" and then it can be fetched from store and made available as a file. My assumption here was from that experience.

If you are referring to the contents of a key file being stored as a string in a cred vault, one has to consider if the vault can store the contents of the key file with formatting. In FIT, the vault cannot, so one has to reformat the file to make sure it is valid.

Reply:
We're presuming a common setup here (credentials in strings from env) for v0 rather than solving for weird scenarios.

If you have a weird scenario that supports file but not strings, you can, in your CI, set env from a file if you want.  Or do the entire auth process from the CI and then testkit will just use the username you passed in (hopefully that can be a string?)

Reply:
I disagree that Jenkins setup is out of the norm and we should consider it as a viable CI/CD for customers to use.

Reply:
@shetzel I think the disconnect is what we define as the interface for testkit WRT how secrets are presented.

Should CI/CD present secrets that ned to be in files as a file or as a string?

Reply:
Reading these comments for the first time confused me at first. Just to be very clear, it isn't being copied. A file is being written from the contents of the env var.

With that said, I think two things.

1. For now, testkit is primarily being developed for our use case and we are 100% in on circleci moving forward. I think we should focus on the circleci happy path and incorporate others as needed.
1. I actually like the file contents in the envvar because it is easier setup. If this is what we do though, we should update our CI examples.